File: easy/0868_binary_gap/solution.py
class Solution:
    def binaryGap(self, n: int) -> int:
        nBin = bin(n)

        left = nBin.find("1")
        right = nBin.find("1", left + 1) if left != -1 else -1

        maxGap = 0

        while right != -1:
            maxGap = max(maxGap, right - left)

            left = right
            right = nBin.find("1", left + 1)

        return maxGap

File: easy/0868_binary_gap/solution.cpp
#include <bitset>
#include <string>

class Solution {
public:
  int binaryGap(int n) {
    std::string nBin = std::bitset<32>(n).to_string();

    size_t left = nBin.find('1');
    size_t right = left != std::string::npos ? nBin.find('1', left + 1)
                                             : std::string::npos;

    int maxGap = 0;

    while (right != std::string::npos) {
      maxGap = std::max(maxGap, static_cast<int>(right - left));

      left = right;
      right = nBin.find('1', left + 1);
    }

    return maxGap;
  }
};
